### DISCUSSION TITLE
DietPi-Software | Prometheus node exporter

### DISCUSSION THREAD

**Discussion by alexiri** [MichaIng/DietPi#5549] [Software Request] [answered]
#### Formal software information
- Software title | Prometheus node exporter
- Short description | Prometheus exporter for hardware and OS metrics exposed by *NIX kernels, written in Go with pluggable metric collectors.
- Official URL (if available) | https://prometheus.io/
- GitHub URL (if available) | https://github.com/prometheus/node_exporter
- (Official) install documentation | https://prometheus.io/docs/guides/node-exporter/

#### Are there similar/alternative software titles available with DietPi-Software?
- The software in [System Stats & Management](https://dietpi.com/docs/software/system_stats/) also monitors the machine.

#### What makes your requested software better than the above solutions, if available?
- Prometheus node exporter is a lightweight daemon that only exposes metrics. Those metrics must then be scraped and aggregated by Prometheus (normally running on a separate system) and then they can be graphed by Grafana. This solution allows multiple DietPi (or other) systems to be monitored from a central location.
- Node exporter on it's own will already provide very useful metrics, but it can be enhanced with RPI-specific information with a project like [raspberrypi_exporter](https://github.com/fahlke/raspberrypi_exporter).

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?

- Installation is pretty simple, there's not much to do.

#### Can you provide the installation steps that you would suggest DietPi-Software to do?

1. Download the latest compatible release from: https://github.com/prometheus/node_exporter/releases
2. Untar and move the `node_exporter` binary to `/usr/local/bin` (or similar)
3. `chmod +x` the binary
4. Add a `node_exporter` user
5. Create a working directory owned by the user: `mkdir /var/lib/node_exporter && chown -R node_exporter:node_exporter /var/lib/node_exporter`
6. Create and enable a simple systemd unit file to run the daemon
7. If also installing `raspberrypi_exporter`, copy what [their installer](https://github.com/fahlke/raspberrypi_exporter/blob/master/installer.sh) does.

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.

- Yes, I could help out with this

**Comment by Joulinar** [comment]
I moved the request die GitHub discussions. Vote for this software implementation with the ⬆️ button on this post.

**Comment by Joulinar** [comment]
But this would require in addition a Prometheus server to be able to use the Prometheus node exporter?

https://github.com/prometheus/prometheus

**Reply by alexiri** [reply]
Yes*, though the Prometheus server doesn't necessarily run on the same machine. The common setup is a central Prometheus server scraping metrics endpoints (which is what the node exporter provides) from multiple machines to aggregate metrics.

*Technically, the node exporter provides a well-known text-based output and it could be used without a Prometheus server, but I can't imagine why somebody would want to.

By the way, I've done an initial implementation of this at #5551.

**Comment by Joulinar** [comment, marked as answer]
Will be part of DietPi 8.6
